### PR TITLE
Set timezone based on TZ environment variable

### LINKED ISSUE #123: Unable to change timezone in jordan/icinga2 docker image
Hi All,

I am trying to test a monitoring tool using icinga2 for that I have used the docker image "jordan/icinga2" (https://hub.docker.com/r/jordan/icinga2/). The installation was good and am able to monitor my servers/hosts. But the issue I'm facing is timezone, the icinga2 showing the UTC timezone where I want to use IST (Asia/Kolkata)......

Changed the timezone from php.ini (/etc/php/7.0/cli/php.ini) but no luck
Can anyone please let me know the possible workaround in this issue?
Thanks in advance

### PR DESCRIPTION
**Pull request by magnusbaeck**
Setting the TZ environment variable won't affect the timezone of the processes started by the supervisor since it trims the environmentpassed to the started child processes.

The proper way to set the timezone in Debian is to write the timezone name to /etc/timezone, adjust the /etc/localtime symlink, and finally reconfigure the tzdata package. We do this in an early setup script.

Fixes #123

### CODE REVIEW

File: content/opt/setup/11-timezone [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+. /opt/helpers
+
+[ -n "${TZ-}" ] || exit 0
+
+echo "=> Setting timezone to ${TZ}"
+
+TZFILE="/usr/share/zoneinfo/${TZ}"
+if [ ! -f ${TZFILE} ]; then
```
Comment (bebehei) [inline content/opt/setup/11-timezone]: This explodes, when TZ contains a space. Please use quotes _everywhere_.

File: content/opt/setup/11-timezone [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	echo "Timezone ${TZ} not available, using default timezone"
+	exit 0
+fi
+
+# As of Debian Jessie it's not enough to update /etc/timezone; the
```
Comment (bebehei) [inline content/opt/setup/11-timezone]: I never heard of using `/etc/timezone`.  Why is this necessary?

Comment (bebehei) [reply content/opt/setup/11-timezone]: The only usable info I found is this thread on stackexchange, which says that java may use it: https://unix.stackexchange.com/a/452566

libc just requires `/etc/localtime`.

Comment (bebehei) [reply content/opt/setup/11-timezone]: Tested in a container, this suffices:

```
 ln -sf "${TZFILE}" /etc/localtime
 dpkg-reconfigure -f noninteractive tzdata
```

`tzdata` reconfiguration listens to `/etc/localtime` setting and updates `/etc/timezone` accordingly. No need to put code in this script and waste brainpower in a few years to ask oneself: Why is this here?

File: content/opt/setup/11-timezone (hunk 1 above)
Comment (magnusbaeck) [reply content/opt/setup/11-timezone]: Done.

File: content/opt/setup/11-timezone (hunk 2 above)
Comment (magnusbaeck) [reply content/opt/setup/11-timezone]: Not sure why I thought it was necessary to update /etc/timezone. tzdata's reconfigure script does it for you as long as you set the symlink.

**Comment by magnusbaeck** [comment]
Thanks for the swift review! Branch updated.

**Comment by jjethwa** [comment]
Thanks @magnusbaeck for the PR and thanks @bebehei for the review 😄
